echopype/convert/api.py

```diff
@@ -420,24 +442,58 @@ def open_raw(
     # Check file extension and existence
     file_chk, xml_chk = _check_file(raw_file, sonar_model, xml_path, storage_options)
 
+    # Ensure offload_to_zarr is 'auto', if it is a string
+    if isinstance(offload_to_zarr, str) and offload_to_zarr != "auto":
+        raise ValueError("offload_to_zarr must be a bool or equal to 'auto'.")
+
     # TODO: the if-else below only works for the AZFP vs EK contrast,
     #  but is brittle since it is abusing params by using it implicitly
     if SONAR_MODELS[sonar_model]["xml"]:
         params = xml_chk
     else:
         params = "ALL"  # reserved to control if only wants to parse a certain type of datagram
 
+    # obtain dict associated with directly writing to zarr
+    dgram_zarr_vars = SONAR_MODELS[sonar_model]["dgram_zarr_vars"]
+
     # Parse raw file and organize data into groups
     parser = SONAR_MODELS[sonar_model]["parser"](
-        file_chk, params=params, storage_options=storage_options
+        file_chk, params=params, storage_options=storage_options, dgram_zarr_vars=dgram_zarr_vars
     )
+
     parser.parse_raw()
+
+    # code block corresponding to directly writing parsed data to zarr
+    if offload_to_zarr and (sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]):
+
+        # Determines if writing to zarr is necessary and writes to zarr
+        p2z = SONAR_MODELS[sonar_model]["parser2zarr"](parser)
+
+        # TODO: perform more robust tests for the 'auto' heuristic value
+        if offload_to_zarr == "auto" and p2z.write_to_zarr(mem_mult=0.4):
+            p2z.datagram_to_zarr(max_mb=max_zarr_mb)
+        elif offload_to_zarr is True:
+            p2z.datagram_to_zarr(max_mb=max_zarr_mb)
+        else:
+            del p2z
+            p2z = Parsed2Zarr(parser)
+            if "ALL" in parser.data_type:
+                parser.rectangularize_data()
+
+    else:
+        p2z = Parsed2Zarr(parser)
+        if (sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]) and (
+            "ALL" in parser.data_type
+        ):
+            parser.rectangularize_data()
+
     setgrouper = SONAR_MODELS[sonar_model]["set_groups"](
         parser,
         input_file=file_chk,
         output_path=None,
         sonar_model=sonar_model,
         params=_set_convert_params(convert_params),
+        parser2zarr_obj=p2z,
```

Comment:
What's the difference between `parsed2zarr` and `parser2zarr`. The inconsistency in tenses is confusing to me.. I assume that they're the same? However this attribute doesn't make sense, I see it as wanting `parser2zarr` object, but there is no such thing, but there is however `parsed2zarr` object.

Reply:
@lsetiawan that's a good catch! I will correct all occurrences to `parsed2zarr`.